### ISSUE TITLE
ViewTransition leads to STATUS_BREAKPOINT on Chrome, (2) overrides html attributes

### ISSUE THREAD

**Issue by Pterygoidien** [withastro/astro#8341]
### Astro Info

```block
Astro version            v3.0.3
Package manager          npm
Platform                 win32
Architecture             x64
Adapter                  @astrojs/node
Integrations             @astrojs/tailwind, @astrojs/mdx, @astrojs/solid-js, @astrojs/prefetch
```

### If this issue only occurs in one browser, which browser is a problem?

Chrome

### Describe the Bug

With the release of Astro version 3.0,  I have tried to implement the new feature "View Transitions" by putting the component in my header. It works on Firefox, but on Chrome, it generates an error "STATUS_BREAKPOINT" when navigating to different pages. Most pages have a common layout (BaseLayout, which can be extended to add wrappers when importing an MD(X) content in the page).  

Moreover, the view transition overrides already existing attributes in my html tag (such as lang, and custom attributes for CSS such as data-theme, etc.). 
![2023-09-01 03_08_54-Accueil _ v-lock](https://github.com/withastro/astro/assets/93955154/5e1ee741-f37f-4ad0-b770-d4c15a21cf38)

### What's the expected result?

I'd like workarounds to avoid HTML attributes being overriden since the whole styling depends on attributes on the <html> tag, and now how to avoid STATUS_BREAKPOINT in Chrome. 

### Link to Minimal Reproducible Example

https://github.com/Pterygoidien/astro-vlock

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by xdlrt** [comment]
I  have same issue with custom attributes in html like data-theme, for dark mode, I wonder how to solve this problem

**Comment by matthewp** [comment]
@xdlrt you probably want the Astro:after-swap event.

**Comment by matthewp** [comment]
@Pterygoidien your link is a 404. Can you provide a reproduction please? Thank you.

**Comment by xdlrt** [comment]
> @xdlrt you probably want the Astro:after-swap event.

this event works for me, thanks

**Comment by Pterygoidien** [comment]
The link to my github seems valid, tried it again. 
Here it is again : https://github.com/Pterygoidien/astro-vlock

**Comment by matthewp** [comment]
@Pterygoidien can you try again with `3.1.1`? Thanks

**Comment by matthewp** [comment]
I think this was fixed in 3.1.1. Going to close, please file a new issue if you still have the problem.
